Makefile

```diff
@@ -305,6 +313,21 @@ run-proxy:
 run-build-vm: $(BIOS_IMG) $(DEVICETREE_DTB)
 	$(QEMU_SYSTEM) $(QEMU_OPTS) -drive format=qcow2,file=$(BUILD_VM)
 
+run-live-parallels:
+	@[ -d "$(LIVE).parallels" ] || { echo "Please run: make live-parallels"; exit 1; }
+	@if prlctl list --all | grep "$(PARALLELS_VM_NAME)"; then \
+		echo "VM exist, please stop $(PARALLELS_VM_NAME) VM"; \
+		prlctl stop $(PARALLELS_VM_NAME) --kill; \
+		prlctl set $(PARALLELS_VM_NAME) --device-set hdd0 --image $(LIVE).parallels --nested-virt on --adaptive-hypervisor on --cpus $(PARALLELS_CPUS) --memsize $(PARALLELS_MEMORY); \
+	else \
+		prlctl create $(PARALLELS_VM_NAME) --distribution ubuntu --no-hdd --dst $(DIST)/ ; \
```

Comment:
I'm still not convinced this block should be here instead of part of `$(LIVE).parallels` -- after all these steps are only run once -- it seems like they should be treated as an extension of image creation, not the run step.

Reply:
LIVE.parallels - creating one HDD. Which you can upload Parallels Remote Server or ship it to other places if you don't need to create VM(can be a good part of CI and releasing process). Plus its takes several minutes to build raw first.

run-live-parallels - lightweight operation and take several seconds to build VM and attach HDD with eve. The main reason why I separate them is parallels register the state of VM to show in Parallels Desktop panel. So it builds VM.pvm + registers the state. This is unnecessary if we need only a parallels image.

Reply:
> LIVE.parallels - creating one HDD. Which you can upload to Parallels Remote Server or ship it to other places if you don't need to create VM(can be a good part of CI and releasing process). Plus, it takes several minutes to build raw first.

Ok. Fair enough

> run-live-parallels - lightweight operation and takes several seconds to build VM and attach HDD with eve. The main reason why I separate them is parallels register the state of VM to show in Parallels Desktop panel. So it builds VM.pvm + registers the state. This is unnecessary if we need only a parallels image.

This then need to be made more robust. Here's what I'm getting after running `make live-parallels && make run-live-parallels` on a clean repo (no pre-existing dist):

```
VM exist, please stop live VM
Stopping the VM...
Failed to stop the VM: Unable to complete the action for "%1". No virtual machine configuration file could be found in the location "%2".
Failed to apply settings: Unable to complete the action for "live". No virtual machine configuration file could be found in the location "/Users/roman/src/lfe/gh/dist/amd64/live.pvm/config.pvs".
Failed to configure the virtual machine.
make: *** [run-live-parallels] Error 255
```

I think that has to do with the fact that Parallels somehow knows of the VM that I created before and thus fails to register a new one. Note that it is totally fine to keep replacing an older VM with a new one - it just has to work reliably.

Reply:
Solve that we able by `prlctl unregister <vm_ID|vm_name>` in make clean. Before deleting  `$(DIST)` where  vm_name.pvm stored. So we need to `prlctl unregister vm_name` and after delete vm_name.pvm files(this is a recommendation from parallels people). This solution can not cover cases if the developer somehow deleted by` rm -rf $(DIST)`. However, Parallels still will have registered `vm_name`.  Another way to make many VMs with different names( vm_name_timestamp or ask developer before creating)

Reply:
Yeah -- I don't think we can ever rely on `make clean` being a requirement for correct working of anything. Also having multiple names for VMs is probably an overkill at this point. Here's what I'm seeing on my laptop after `rm -rf dist` (which happens all the time):

```
$ prlctl list -a
UUID                                    STATUS       IP_ADDR         NAME
{8024ca99-a99c-4ddf-bf52-109470b050df}  suspended    -               Debian GNU Linux 10
{fdae88a4-a9c7-416d-928f-c9fc85425161}  suspended    -               PLC4X Demo (Windows 10)
{e8aa0d52-eecf-4d65-a938-56cd4780674f}  invalid      -               live
{c1d20ce4-f685-40bd-a3ac-f6d69f830537}  suspended    -               macOS
```

So I suggest you simply look for `live` VMs (I actually suggest you call them `EVE Live`) in the invalid state and silently take it as a hint to clean up that state and register a new VM.

Will this work?

Reply:
Yes, this one will work as well.